godot-macros/src/class/godot_api.rs

```diff
@@ -8,10 +8,30 @@
 use proc_macro2::TokenStream;
 
 use crate::class::{transform_inherent_impl, transform_trait_impl};
-use crate::util::bail;
+use crate::util::{bail, KvParser};
 use crate::ParseResult;
 
-pub fn attribute_godot_api(input_decl: venial::Item) -> ParseResult<TokenStream> {
+use quote::quote;
+
+fn parse_inherent_impl_attr(meta: TokenStream) -> Result<super::InherentImplAttr, venial::Error> {
+    // Hack because venial doesn't support direct meta parsing yet.
+    let input = quote! {
+        #[godot_api(#meta)]
+        fn func();
+    };
```

Comment:
Could you reuse this here? Maybe extract in a function? (without `and_then`)

https://github.com/godot-rust/gdext/blob/8a62088d8526ec9ad4722211377d20221f5a442f/godot-macros/src/lib.rs#L964-L972

Reply:
hmm, I do feel leaving it duplicated is clearer that trying to DRY it out, because at the call site, you don't really know what exactly it does unless you look into the implementation.

Since you're the boss, I've created a util function anyway

Reply:
I've added it in util.rs, lmk if there's a better place for it